custom_components/aprilaire/humidifier.py

```diff
@@ -0,0 +1,225 @@
+        """Return the target humidity."""
+        setpoint = self.coordinator.data.get(Attribute.DEHUMIDIFICATION_SETPOINT)
+        """Don't set last target humidity if off."""
+        self._last_is_on = setpoint != 0
+        self._last_target_humidity = setpoint if self._last_is_on else self._last_target_humidity
+        return self._last_target_humidity
+
+    @property
+    def is_on(self) -> bool:
+        """Return true if on."""
+        return self.target_humidity != 0
+
+    @property
+    def mode(self) -> str | None:
+        """Get dehumidifier mode."""
+
+        mode = self.coordinator.data.get(Attribute.MODE)
+
+        if mode == 4:
+            return MODE_VACATION
+
+        return MODE_MANUAL
+
+    @property
+    def current_humidity(self) -> int | None:
+        """Get current humidity."""
+        return self.coordinator.data.get(
+            Attribute.INDOOR_HUMIDITY_CONTROLLING_SENSOR_VALUE
+        )
+
+    async def async_set_humidity(self, humidity: int) -> None:
+        """Set the target dehumidification setpoint."""
+        await self.coordinator.client.set_dehumidification_setpoint(humidity)
+
+    async def async_turn_on(self, **kwargs: Any) -> None:
+        """Turn on the dehumidifier."""
+        await self.async_set_humidity(self._last_target_humidity)
+
+    async def async_turn_off(self, **kwargs: Any) -> None:
+        """Turn off the dehumidifier."""
+        await self.async_set_humidity(0)
+
+    async def async_set_mode(self, mode: str) -> None:
+        """Set the mode of the thermostat."""
+        await self.coordinator.client.read_scheduling()
+
+class AprilaireHumidifier(BaseAprilaireEntity, HumidifierEntity):
+    """Humidifier entity for Aprilaire."""
+
+    _attr_supported_features = HumidifierEntityFeature.MODES
+    _attr_device_class = HumidifierDeviceClass.HUMIDIFIER
+    _attr_min_humidity = 10
+    _attr_max_humidity = 50
+
+    def __init__(self, coordinator: AprilaireCoordinator) -> None:
+        self._last_target_humidity = None
+        self._last_is_on = None
+
+    @property
+    def action(self) -> HumidifierAction | None:
+        """Return the current action."""
+        
+        status = self.coordinator.data.get(
+            Attribute.HUMIDIFICATION_STATUS
+        )
+
+        if status == 0 or status == 1:
+            return HumidifierAction.IDLE
+        
+        if status == 2:
+            return HumidifierAction.HUMIDIFYING
+        
+        if status == 3:
+            return HumidifierAction.OFF
+        
+        raise ValueError(f"Unsupported humidification status {status}")
+
+    @property
+    def target_humidity(self) -> int:
+        """Return the target humidity."""
+        setpoint = self.coordinator.data.get(Attribute.HUMIDIFICATION_SETPOINT)
+        """Don't set last target humidity if off."""
+        self._last_is_on = setpoint != 0
+        self._last_target_humidity = setpoint if self._last_is_on else self._last_target_humidity
```

Comment:
@nberardi I think I see where you're going with this, which is storing the last humidity so that if the device is turned off and then on, that it uses the last humidity to turn it back on. This has an edge case which you didn't consider which is that if the integration starts with the device off (eg humidity == 0) then it won't be able to turn on. Since there's no way for us to know the previous setpoint in that case, I think we need to set a reasonable default humidity to restore to in that case.